File: C/Colleen/Colleen.c
#include <stdio.h>

static char	*get_code()
{
	char	*code = "#include <stdio.h>%1$c%1$cstatic char	*get_code()%1$c{%1$c	char	*code = %3$c%4$s%3$c;%1$c%1$c	return code;%1$c}%1$c%1$c/*%1$c	This is a comment.%1$c*/%1$cint			main(void)%1$c{%1$c	/*%1$c		And this is another comment.%1$c	*/%1$c	printf(get_code(), '%2$cn', '%2$c%2$c', '%3$c', get_code());%1$c	return 0;%1$c}%1$c";

	return code;
}

/*
	This is a comment.
*/
int			main(void)
{
	/*
		And this is another comment.
	*/
	printf(get_code(), '\n', '\\', '"', get_code());
	return 0;
}

File: Rust/Colleen.rs
macro_rules! code_fmt {
	() => (r#"macro_rules! code_fmt {{{0}	() => (r#{1}{2}{1}#){0}}}{0}{0}fn print_code() {{{0}	print!(code_fmt!(), {1}\n{1}, r#{1}{1}{1}#, code_fmt!());{0}}}{0}//{0}// This is a comment.{0}//{0}{0}fn main() {{{0}	//{0}	// And this is another comment.{0}	//{0}	print_code();{0}}}{0}"#)
}

fn print_code() {
	print!(code_fmt!(), "\n", r#"""#, code_fmt!());
}
//
// This is a comment.
//

fn main() {
	//
	// And this is another comment.
	//
	print_code();
}
